ovh/resource_domain_name_servers.go

```diff
@@ -0,0 +1,220 @@
+	Status        string `json:"status,omitempty"`
+	TodoDate      string `json:"todoDate,omitempty"`
+}
+
+type OvhDomainNameServer struct {
+	Id       int64  `json:"id,omitempty"`
+	Host     string `json:"host,omitempty"`
+	Ip       string `json:"ip,omitempty"`
+	IsUsed   bool   `json:"isUsed,omitempty"`
+	ToDelete bool   `json:"toDelete,omitempty"`
+}
+
+type OvhDomainNameServers struct {
+	Id      string                `json:"id,omitempty"`
+	Domain  string                `json:"domain,omitempty"`
+	Servers []OvhDomainNameServer `json:"servers,omitempty"`
+}
+
+func (r *OvhDomainNameServers) String() string {
+	domains := make([]string, 0)
+	for _, server := range r.Servers {
+		domains = append(domains, fmt.Sprintf(
+			"server[id: %v, host: %s, ip: %s, isUsed: %v, toDelete: %v]",
+			server.Id,
+			server.Host,
+			server.Ip,
+			server.IsUsed,
+			server.ToDelete,
+		))
+	}
+	return fmt.Sprintf(
+		"nameservers[id: %v, domain: %s, servers: [%s]]",
+		r.Id,
+		r.Domain,
+		strings.Join(domains, ", "),
+	)
+}
+
+func resourceOvhDomainNameServersImportState(
+	d *schema.ResourceData,
+	meta interface{}) ([]*schema.ResourceData, error) {
+	givenId := d.Id()
+	d.SetId(givenId)
+	d.Set("domain", d.Id())
+	results := make([]*schema.ResourceData, 1)
+	results[0] = d
+	return results, nil
+}
+
+func resourceOvhDomainNameServers() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceOvhDomainNameServersCreate,
+		Read:   resourceOvhDomainNameServersRead,
+		Update: resourceOvhDomainNameServersCreate, // Update is the same as create
+		Delete: resourceOvhDomainNameServersDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceOvhDomainNameServersImportState,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"domain": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"servers": {
+				Type:     schema.TypeList,
+				Required: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"host": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"ip": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Default:  "",
+						},
+					},
+				},
+				MinItems: 1,
+			},
+		},
+	}
+}
+
+func resourceOvhDomainNameServersCreate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	serviceName := d.Get("domain").(string)
+
+	// Servers to put
+	servers := &OvhDomainNameServerUpdate{
+		NameServers: make([]OvhDomainNameServer, 0),
+	}
+	// Loop due to rename of the field in the API
+	for _, server := range d.Get("servers").([]interface{}) {
+		s := server.(map[string]interface{})
+		servers.NameServers = append(servers.NameServers, OvhDomainNameServer{
+			Host: s["host"].(string),
+			Ip:   s["ip"].(string),
+		})
+	}
+	log.Printf("[DEBUG] OVH Record create configuration: %#v", servers)
+	err := config.OVHClient.Post(fmt.Sprintf("/domain/%s/nameServers/update", serviceName), servers, nil)
```

Comment:
```suggestion
	err := config.OVHClient.Post(fmt.Sprintf("/domain/%s/nameServers/update", url.PathEscape(serviceName)), servers, nil)
```

This is to prevent any path traversal injection